src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const delProp of action.keysToRemove) {
+          delete state[delProp];
+        }
+        break;
+
+      case 'clear':
+        for (const delAllProp in state) {
```

Comment:
same here, you should use proper names for variables. The name of the variable should tell what is stored inside